### PR TITLE
DOM Manipulation and Events: Update lists per layout style guide

### PR DESCRIPTION
**Pull request by adityax4**
## Because

The Markdown files should be updated as per layout style guide for better readability.

## This PR

- Updated lists per layout style guide.
- Added lazy numbering in lists.
- Removed extra unnecessary spaces.

## Issue

Related to #27119

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/DOM_manipulation_and_events.md [hunk 1]
```diff
@@ -13,20 +13,17 @@ This section contains a general overview of topics that you will learn in this l
 - Explain the difference between a "nodelist" and an "array of nodes".
 - Explain what "bubbling" is and how it works.
 
-### DOM - Document Object Model
-
+### DOM Document Object Model
```
Comment (thatblindgeye) [inline foundations/javascript_basics/DOM_manipulation_and_events.md:16]: Let's just update this to remove DOM:

```suggestion
### Document Object Model
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Comment by adityax4** [comment]
hey @thatblindgeye, there's a link error that asks that Link fragments should be valid.
[Context: "[What is the DOM?](#dom-document-object-model)"]
then, this also needs to be updated to:
[What is the DOM?](#document-object-model)
but in html file the <a href attribute is #dom-document-object-model

that's why i removed the hyphen initially from the heading to resolve this lint error.
what to do next? guide please.

**Comment by thatblindgeye** [comment]
@adityax4 you can update that knowledge check url to `#document-object-model`

**Comment by adityax4** [comment]
> @adityax4 you can update that knowledge check url to `#document-object-model`

@thatblindgeye done, everything's good now :)

**Review by thatblindgeye — APPROVED** [review]
💯 Thank you!

**Pull request merged**